### ISSUE TITLE
Identify Components for a v8 Promotion

### ISSUE THREAD

**Issue by myvuuu** [Workday/canvas-kit#1652]
## 💡 Suggestion

Canvas Kit is wanting to promote components that have been in `labs` or `preview`. Some of these components are in a stable state to be promoted while others either need some work or there's future plans to promote them in a later version. This outline serves as a suggestion to which components should be promoted in `v8`.

Fixes: [https://github.com/Workday/canvas-kit/issues/1194](https://github.com/Workday/canvas-kit/issues/1194)

### Guide for promotion

✅   Good candidate for promotion

🚫   Should not be promoted

💁   Need more info / I'm not sure

**Components to be promoted:** `ComponentStatesTable, Breadcrumbs`

## Preview Components

- **Breadcrumbs** ✅ / 💁
    - **Should be promoted?**
        - We need to update the component to use the latest list system
        - Open issues: [https://github.com/Workday/canvas-kit/issues/1314](https://github.com/Workday/canvas-kit/issues/1314), [https://github.com/Workday/canvas-kit/issues/1070](https://github.com/Workday/canvas-kit/issues/1070)
- **Color Picker** 🚫
    - **Should be promoted?**
        - This component will be replaced by the current work in https://github.com/Workday/canvas-kit/pull/1262
- **Inputs [FormField, Textarea, Text Input] (Rusty's Work)** 💁 / ✅
    - **Should be promoted?** 💁
        - I think we had questions about some of the api choices
        - Overall this will have to be updated when we tackle checkbox and radio inputs as well
        - Evaluate form field and `as`
        - Add fieldset component?
        - Needs more discussion
        - Spike cc: Team
- **Menu** 💁
    - **Should be promoted?**
        - Nicholas did a lot of work with the list system and that was merged into main
        - Should we soft deprecate this component in favor for the one in main?
        - Here’s the work that added the list system https://github.com/Workday/canvas-kit/pull/1547
        - [https://github.com/Workday/canvas-kit/issues/1070](https://github.com/Workday/canvas-kit/issues/1070),
        - https://github.com/Workday/canvas-kit/issues/1213
- **Select**  ✅
    - **Should be promoted?**
        - Open issues: [https://github.com/Workday/canvas-kit/issues/780](https://github.com/Workday/canvas-kit/issues/780), [https://github.com/Workday/canvas-kit/issues/607](https://github.com/Workday/canvas-kit/issues/607),
        - I think this component is in a pretty solid state, it does have a few open issues but i think it can be promoted
        - Leave in preview, update one in main?
        - wait on combobox
        - v9? after work with in v8
- **Side Panel**  ✅
    - **Should be promoted?**
        - I think we were already discussing about promoting this
        - The only piece missing is converting to compound but not all of our components in main are compound
        - We should deprecate what is in main and promote this one
        - Soft deprecate drawer

## Lab Components

- **Combobox** 💁 / ✅
    - **Should be Promoted?**
        - This component still needs to be converted to a compound component
        - It uses a text input, maybe this should use the text input from preview and be moved to preview after conversion?
        - Open issues: [https://github.com/Workday/canvas-kit/issues/1174](https://github.com/Workday/canvas-kit/issues/1174), [https://github.com/Workday/canvas-kit/issues/1171](https://github.com/Workday/canvas-kit/issues/1171), [https://github.com/Workday/canvas-kit/issues/1122](https://github.com/Workday/canvas-kit/issues/1122)
        - https://github.com/Workday/canvas-kit/issues/1669
            - **Questions**
                - I think before promoting we should: Convert to compound component, address open issues, and move to preview?
- **useThemedRing, useThemeRTL, ComponentStatesTable**
    - We can promote `ComponentStatesTable` so there’s no dependency from main on this
    - `useThemeRTL` if we’re dropping IE11 in v8 can we use logical properties and remove this?
    - `useThemedRing` not much has changed in this function and it is mostly used by preview and lab components
- **Drawer** 🚫. SOFT DEPRECATE
- **Should be Promoted?**
    - [https://github.com/Workday/canvas-kit/issues/1625](https://github.com/Workday/canvas-kit/issues/1625)
- **Search Form** 💁
    - **Should be Promoted?**
        - The component says that it would be used in a context of a header and the header is deprecated
        - This a nice to have component, again, maybe a component that fits in our examples instead?
        - There's no open issues for it and only one team uses this component
    
    ## Discussion points
    
    - Communicate what we're promoting
    - Figure out deprecation process
    - Discuss future work on promotion for other components

**Comment by mannycarrera4** [comment]
Quick summary:

### Components to be Soft Deprecated
Issue for soft deprecation: https://github.com/Workday/canvas-kit/issues/1712
